src/GraphQL/Types/Schema.cs

```diff
@@ -82,9 +82,12 @@ public void EnsureLookup()
             {
                 _lookup = new GraphTypesLookup();
 
-                var ctx = new TypeCollectionContext(ResolveType, (name, graphType) =>
+                var ctx = new TypeCollectionContext(ResolveType, (name, graphType, context) =>
                 {
-                    _lookup[name] = graphType;
+                    if (_lookup[graphType.Name] == null)
```

Comment:
I think we need to use `name` here, which is passed in, and not `graphType.Name`.  Part of when this code runs is the type may actually not have a `Name` populated yet, or is not a proper name.  This is something I don't like and would like to change.